pkg/query/endpointset_test.go

```diff
@@ -269,13 +271,54 @@ func (e *testEndpoints) CloseOne(addr string) {
 	delete(e.srvs, addr)
 }
 
+func TestTruncateExtLabels(t *testing.T) {
```

Comment:
This test looks great, but we still did not test all edge cases. I play with your code and I reproduced one bug using `"{abcd}"` input....

Reply:
![image](https://user-images.githubusercontent.com/6950331/196931808-13d147f7-0e5b-4fc1-bb81-98dd54284a45.png)